crates/forge-runner/src/running/with_config.rs

```diff
@@ -0,0 +1,135 @@
+use crate::{
+    function_args,
+    package_tests::{
+        raw::TestTargetRaw,
+        with_config::{TestCaseWithConfig, TestTargetWithConfig},
+        TestDetails,
+    },
+    running::config_run::get_config_for_test_case,
+};
+use anyhow::{anyhow, Result};
+use cairo_felt::Felt252;
+use cairo_lang_sierra::{
+    extensions::core::{CoreLibfunc, CoreType},
+    ids::{ConcreteTypeId, GenericTypeId},
+    program::TypeDeclaration,
+    program_registry::ProgramRegistry,
+};
+use cairo_lang_sierra_type_size::get_type_size_map;
+use cairo_lang_utils::unordered_hash_map::UnorderedHashMap;
+use std::{collections::HashMap, sync::Arc};
+use universal_sierra_compiler_api::compile_sierra_to_casm;
+
+pub fn test_target_with_config(test_target_raw: TestTargetRaw) -> Result<TestTargetWithConfig> {
+    let funcs: HashMap<_, _> = test_target_raw
+        .sierra_program
+        .program
+        .funcs
+        .iter()
+        .map(|f| (f.id.id, f))
+        .collect();
+    let type_declarations: HashMap<_, _> = test_target_raw
+        .sierra_program
+        .program
+        .type_declarations
+        .iter()
+        .map(|f| (f.id.id, f))
+        .collect();
+
+    let types: HashMap<_, _> = test_target_raw
+        .sierra_program
+        .program
+        .type_declarations
+        .iter()
+        .map(|ty| (ty.id.id, ty))
+        .collect();
+
+    let casm_program = Arc::new(compile_sierra_to_casm(
+        &test_target_raw.sierra_program.program,
+    )?);
+
+    let sierra_program_registry =
+        ProgramRegistry::<CoreType, CoreLibfunc>::new(&test_target_raw.sierra_program.program)?;
+    let type_size_map = get_type_size_map(
+        &test_target_raw.sierra_program.program,
+        &sierra_program_registry,
+    )
+    .ok_or_else(|| anyhow!("can not get type size map"))?;
+
+    let default_executables = vec![];
+    let executables = test_target_raw
+        .sierra_program
+        .debug_info
+        .executables
+        .get("snforge_internal_test_executable")
+        .unwrap_or(&default_executables);
+
+    Ok(TestTargetWithConfig {
```

Comment:
This fragment until the end of the file needs to be refactored because it's unreadable and it's hard to tell what's happening here

Reply:
Why was this marked as resolved if the file was not changed?